src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs

```diff
@@ -195,7 +198,41 @@ impl VirtSandbox {
         // * spec details: https://github.com/opencontainers/runtime-spec/blob/c1662686cff159595277b79322d0272f5182941b/config.md#createruntime-hooks
         let mut create_runtime_hook_states = HookStates::new();
         create_runtime_hook_states.execute_hooks(create_runtime_hooks, Some(st.clone()))?;
+        Ok(())
+    }
+
+    // store_guest_details will get the information from the guest OS, like memory block size, agent details and is memory hotplug probe support
+    async fn store_guest_details(&self) -> Result<()> {
+        // get the information from agent
+        let guest_details = self
+            .agent
+            .get_guest_details(GetGuestDetailsRequest {
+                mem_block_size: true,
+                mem_hotplug_probe: true,
+            })
+            .await
+            .context("failed to store guest details")?;
 
+        // set memory block size
+        self.hypervisor
+            .set_guest_memory_block_size(guest_details.mem_block_size_bytes as u32)
+            .await;
+
+        // set memory hotplug probe
+        if guest_details.support_mem_hotplug_probe {
+            self.hypervisor
+                .set_capabilities(CapabilityBits::GuestMemoryHotplugProbe)
```

Comment:
why do we need this checking? set cap and check it?

Reply:
This is because we need to know if the hypervisor support memory hotplug. we get `support_mem_hotplug_probe ` from agent response, and it is to determine the capability of memory hotplug, if it is, we set the capability to hypervisor.